src/common/LogEntry.h

```diff
@@ -111,16 +127,22 @@ WRITE_CLASS_ENCODER_FEATURES(LogEntry)
 
 struct LogSummary {
   version_t version;
+
+  // ---- pre-quincy ----
   // channel -> [(seq#, entry), ...]
   std::map<std::string,std::list<std::pair<uint64_t,LogEntry>>> tail_by_channel;
   uint64_t seq = 0;
   ceph::unordered_set<LogEntryKey> keys;
 
+  // ---- quincy+ ----
+  LRUSet<LogEntryKey> recent_keys;
+  std::map<std::string, pair<uint64_t,uint64_t>> channel_info;
```

Comment:
```suggestion
  struct log_range_t {
    uint64_t first;
    uint64_t last;
  };
  std::map<std::string, log_range_t> channel_info;
```

Reply:
leaving this as first/second to avoid having to write encode/decode etc.